### PR TITLE
Add RMSNorm  (part of parallel_attn_blocks)

### PR DESCRIPTION
**Pull request by lessw2020**
Summary:
This PR adds RMSNorm.  
It is used in parallel_attention_blocks but moving the earlier mono PR for that into smaller chunks for easier review and landing. 

Test plan:
added two units tests:
test_rms_norm_fp32
test_rms_core_algo  (compares F.norm version vs PR version)

Fixes:
Adds requested doc string to RMSNorm from earlier PR review.

### CODE REVIEW

**Review by rohan-varma — APPROVED** [review]
LG! I would just absorb the fp32 type check test into the other one for brevity.

File: torchmultimodal/modules/layers/normalizations.py [hunk 1]
```diff
@@ -45,3 +46,25 @@ def forward(self, x: Tensor) -> Tensor:
             self.eps,
         )
         return output.type_as(x)
+
+
+class RMSNorm(nn.Module):
+    """Root Mean Square layer normalization
+    as proposed in: https://arxiv.org/abs/1910.07467
+
```
Comment (ankitade) [inline torchmultimodal/modules/layers/normalizations.py:54]: i think this is copy paste of llama. can u add a link

File: torchmultimodal/modules/layers/normalizations.py [hunk 2]
```diff
@@ -45,3 +46,25 @@ def forward(self, x: Tensor) -> Tensor:
...
+    params:
+    dim = model size
+    eps = epsilon
```
Comment (ankitade) [inline torchmultimodal/modules/layers/normalizations.py:57]: nit: follow this format https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/models/dalle2/dalle2_decoder.py#L48 (type, default values etc)

File: tests/modules/layers/test_normalizations.py [hunk 3]
```diff
@@ -20,3 +28,45 @@ def test_fp32groupnorm():
...
+    dims = 512
+    x = torch.empty(dims, dtype=torch.float16)
+    norm = RMSNorm(
+        dims,
+    )
+    output = norm(x)
+    assert output.dtype == torch.float32
+
+
+@gpu_test(1)
```
Comment (ankitade) [inline tests/modules/layers/test_normalizations.py:44]: why is gpu required? for fp16 support?

File: tests/modules/layers/test_normalizations.py [hunk 4]
```diff
@@ -20,3 +28,45 @@ def test_fp32groupnorm():
...
+def test_rms_norm_core_algo():
+    """compare RMSNorm with RMSNorm using F.norm version"""
+
+    dims = 1024
+    x = torch.empty(dims, dtype=torch.float16, device="cuda")
+    x_clone = x.clone().detach()
+
+    class RMSNormFunctional(torch.nn.Module):
```
Comment (ankitade) [inline tests/modules/layers/test_normalizations.py:52]: this test is little weirdly written. u should just assert on output values directly (that u hand calculated or got from reference impl)

File: tests/modules/layers/test_normalizations.py [hunk 5]
```diff
@@ -20,3 +28,45 @@ def test_fp32groupnorm():
     norm = Fp32GroupNorm(2, 4)
     output = norm(x)
     assert output.dtype == torch.float16
+
+
+def test_rms_norm_fp32return():
```
Comment (ankitade) [inline tests/modules/layers/test_normalizations.py:33]: why dont we just do 1 test with fp32 and assert on output values?

File: torchmultimodal/modules/layers/normalizations.py (hunk 1 above)
Comment (lessw2020) [reply torchmultimodal/modules/layers/normalizations.py:54]: added

File: torchmultimodal/modules/layers/normalizations.py (hunk 2 above)
Comment (lessw2020) [reply torchmultimodal/modules/layers/normalizations.py]: updated to use similar format

File: tests/modules/layers/test_normalizations.py (hunk 3 above)
Comment (lessw2020) [reply tests/modules/layers/test_normalizations.py:33]: it's not required per se - however for finer precision / lower precision dtypes I have seen times where the PT cpu version impl does not match the gpu version impl.
Since we care about gpu, figured it's better to run this test on gpu. 
Can remove this if it creates to much test overhead?

File: tests/modules/layers/test_normalizations.py (hunk 5 above)
Comment (lessw2020) [reply tests/modules/layers/test_normalizations.py]: my idea was to separate out the tests esp if we add more dtype support (ala fp8) but will merge this into one for brevity.

File: tests/modules/layers/test_normalizations.py (hunk 4 above)
Comment (lessw2020) [reply tests/modules/layers/test_normalizations.py]: ok moved to fixed tensor tests instead with pre-calced results.

File: tests/modules/layers/test_normalizations.py (hunk 3 above)
Comment (lessw2020) [reply tests/modules/layers/test_normalizations.py]: removed the gpu decorator since CI does not necessarily have gpu.

File: tests/modules/layers/test_normalizations.py [hunk 6]
```diff
@@ -20,3 +26,38 @@ def test_fp32groupnorm():
     norm = Fp32GroupNorm(2, 4)
     output = norm(x)
     assert output.dtype == torch.float16
+
+
+def test_rms_norm_core_algo():
```
Comment (ankitade) [inline tests/modules/layers/test_normalizations.py:31]: nit: rename to test_rms_norm

**Review by ankitade — APPROVED** [review]

File: tests/modules/layers/test_normalizations.py (hunk 6 above)
Comment (lessw2020) [reply tests/modules/layers/test_normalizations.py:31]: that makes sense.  I'll resolved in the fb diff.
